model/Makefile

```diff
@@ -25,44 +25,50 @@ install:
 test:
 	poetry run pytest --mpl --mpl-default-tolerance=10
 
-docs: docs/pyrenew_demo.md docs/getting-started.md \
-	docs/example-with-datasets.md docs/extending_pyrenew.md
+# Pattern rule to build a markdown file from a qmd file
+docs/%.md: docs/%.qmd
+	poetry run quarto render $<
+
+# Pattern rule to build a notebook file from a qmd file
+docs/%.ipynb: docs/%.qmd
+	poetry run quarto convert $< --output $@
+
+# Pattern rule to build a Python file from a notebook file
+docs/%.py: docs/%.ipynb
+	sed -i '' 's/"cell_type": "raw"/"cell_type": "markdown"/g' $<
+	poetry run jupyter nbconvert --to python $<
+
+# Automatically get all .qmd files in the docs directory
+QMD_FILES := $(wildcard docs/*.qmd)
+
+# Derive the corresponding markdown files
+MD_FILES := $(QMD_FILES:.qmd=.md)
 
-docs/pyrenew_demo.md: docs/pyrenew_demo.qmd
-	poetry run quarto render docs/pyrenew_demo.qmd
+# Derive the corresponding notebook files
+NOTEBOOK_FILES := $(QMD_FILES:.qmd=.ipynb)
 
-docs/getting-started.md: docs/getting-started.qmd
-	poetry run quarto render docs/getting-started.qmd
+# Derive the corresponding Python files
+PY_FILES := $(NOTEBOOK_FILES:.ipynb=.py)
 
-docs/example-with-datasets.md: docs/example-with-datasets.qmd
-	poetry run quarto render docs/example-with-datasets.qmd
+# Target to build all markdown files
+docs: $(MD_FILES)
```

Comment:
My only question is whether to include `$(NOTEBOOK_FILES)` and `$(PY_FILES)` in the phony `docs` target (versus and/or in addition to creating their own phony targets)

Reply:
As in `make docs` would create the `md`, `ipynb`, and `py` files? And we would have a separate make `docs/md` to just create the `md`s?

Reply:
right. Also might prefer `docs_md`/`docs_py`/`docs_ipynb` to `docs/md` etc for the names of the phony targets to avoid the appearance of subdirectories.